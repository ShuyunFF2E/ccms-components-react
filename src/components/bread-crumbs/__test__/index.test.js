import React from 'react';
import { mount, render } from 'enzyme';
import mountTest from '@tests/shared/mountTest';
import BreadCrumbs from '../index';

describe('BreadCrumbs', () => {
  mountTest(BreadCrumbs);
  let list = null;

  beforeAll(() => {
    list = [
      {
        key: 'home',
        title: '首页'
      },
      {
        key: 'bread-crumbs',
        title: '面包屑'
      }
    ];
  });

  afterAll(() => {
    list = null;
  });

  it('renders correctly', () => {
    const wrapper = render(<BreadCrumbs list={list} />);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.hasClass('cloud-breadcrumbs')).toBeTruthy();
    expect(wrapper.find('li').length).toBe(2);
  })

  it('should be trigger click', () => {
    let activeKey = null;
    const wrapper = mount(<BreadCrumbs list={list} onClick={item => { activeKey = item.key }} />);

    wrapper.find('li').at(0).simulate('click')
    expect(activeKey).toBe('home');

    wrapper.find('li').at(1).simulate('click')
    expect(activeKey).toBe('bread-crumbs');

    activeKey = null;
  })

  it('should be support different size', () => {
    const size = ['large', 'default', 'small'];
    size.forEach(type => {
      const wrapper = mount(<BreadCrumbs list={list} size={type}/>);

      expect(wrapper.find(`.cloud-breadcrumbs-${type}`)).toBeTruthy();
      expect(wrapper.find('li').length).toBe(2);
      wrapper.unmount();
    });
  })
});
