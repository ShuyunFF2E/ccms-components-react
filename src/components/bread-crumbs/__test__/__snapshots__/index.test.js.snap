// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs renders correctly 1`] = `
<ul
  class="cloud-breadcrumbs cloud-breadcrumbs-default"
>
  <li>
    <span
      class="title"
    >
      首页
    </span>
    <span
      class="split"
    >
      /
    </span>
  </li>
  <li>
    <span
      class="title"
    >
      面包屑
    </span>
  </li>
</ul>
`;
